start_admin_sshd.sh

```diff
@@ -1,31 +1,34 @@
-#!/bin/bash
-# This file is part of Bottlerocket.
-# Copyright Amazon.com, Inc., its affiliates, or other contributors. All Rights Reserved.
-# SPDX-License-Identifier: Apache-2.0 OR MIT
+#!/usr/bin/env bash
+
 set -e
 
+declare -r PERSISTENT_STORAGE_BASE_DIR="/.bottlerocket/host-containers/current"
+declare -r SSH_HOST_KEY_DIR="${PERSISTENT_STORAGE_BASE_DIR}/etc/ssh"
+declare -r USER_DATA="${PERSISTENT_STORAGE_BASE_DIR}/user-data"
+
+declare -r LOCAL_USER="ec2-user"
+declare -r USER_SSH_DIR="/home/${LOCAL_USER}/.ssh"
+
+# This is a counter used to verify at least
+# one of the methods below is available.
+declare -i available_auth_methods=0
+
 log() {
   echo "$*" >&2
 }
 
-declare -r local_user="ec2-user"
-declare -r persistent_storage_base_dir="/.bottlerocket/host-containers/current"
-declare -r ssh_host_key_dir="${persistent_storage_base_dir}/etc/ssh"
-declare -r user_data="${persistent_storage_base_dir}/user-data"
-declare -r user_ssh_dir="/home/${local_user}/.ssh"
-available_auth_methods=0
-
-mkdir -p "${user_ssh_dir}"
-chmod 700 "${user_ssh_dir}"
-
 get_user_data_keys() {
     # Extract the keys from user-data json
     local raw_keys
     local key_type="${1:?}"
-    # ? signifies an optional object identifier-index and doesn't return a 'failed to iterate' error
-    # in the event that a key_type is missing. If jq returns nothing, then that is caught by the -z.
-    if ! raw_keys=$(jq --arg key_type "${key_type}" -e -r '.["ssh"][$key_type][]?' "${user_data}") \
-    || [[ -z "${raw_keys}" ]]; then
+    # ? signifies an optional object identifier-index and doesn't return a
+    # 'failed to iterate' error in the event that a key_type is missing.
+    if ! raw_keys=$(jq --arg key_type "${key_type}" -e -r '.["ssh"][$key_type][]?' "${USER_DATA}"); then
```

Comment:
I'm getting to the point where I think we ought to add tests in the very very near future (not this PR; maybe with [BATS](https://github.com/bats-core/bats-core)?) for the scripts in the host containers. The functionality is steadily increasing that having some unit tests for processing functions would be great to ensure expected cases are handled appropriately.

Reply:
I'm a fan of automated testing 🤖